test/parse.js

```diff
@@ -236,6 +236,14 @@ test('parse()', function (t) {
         st.end();
     });
 
+    t.test('parses jquery-param strings', function (st) {
+        // readable = 'filter[0][]=int1&filter[0][]==&filter[0][]=77&filter[]=and&filter[2][]=int2&filter[2][]==&filter[2][]=8'
+        var encoded = 'filter%5B0%5D%5B%5D=int1&filter%5B0%5D%5B%5D=%3D&filter%5B0%5D%5B%5D=77&filter%5B%5D=and&filter%5B2%5D%5B%5D=int2&filter%5B2%5D%5B%5D=%3D&filter%5B2%5D%5B%5D=8';
```

Comment:
any chance we could programmatically construct `encoded` from `readable`, so the comment isn't needed?

Reply:
We could. The OP mentioned this specific encoded string was produced by jQuery, so I intentionally hard-coded that exact input for the test to avoid any other (possibly changing) influences on what was being tested.

Using the readable string also produces the error without the fix, so we could just use that? Downside is we'd no longer be testing the exact `jQuery.param()` output. 

I could just remove the comment.

Reply:
We could. The OP mentioned this specific encoded string was produced by jQuery, so I hard-coded that exact input for the test.

Using the readable string also produces the error without the fix, so we could just use that? Downside is we'd no longer be testing the exact jQuery.param() output.

I could just remove the comment.